src/Bicep.LangServer/Handlers/BicepDeploymentParametersHandler.cs

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Deploy;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    public record BicepDeploymentParametersResponse(List<BicepDeploymentParameter> deploymentParameters, string parametersFileName, string? errorMessage);
+
+    public record BicepDeploymentParameter(string name, string? value, bool isMissingParam, bool isExpression, bool isSecure, ParameterType? parameterType);
+
+    /// <summary>
+    /// Handles getDeploymentParameters LSP request.
+    /// The BicepDeploymentParametersHandler returns information about deployment parameters, parameters file name and error message, if any.
+    /// List of <see cref="BicepDeploymentParameter"/>, included in the response has informtion about the parameter e.g. name , value, if the parameter has
+    /// @secure() decorator, if it's missing default value/is not present in parameters file, is an expression etc
+    /// The above information will be used to display appropriate controls in UI.
+    /// </summary>
+    public class BicepDeploymentParametersHandler : ExecuteTypedResponseCommandHandlerBase<string, string, string, BicepDeploymentParametersResponse>
+    {
+        private readonly ICompilationManager compilationManager;
+
+        public BicepDeploymentParametersHandler(
+            ICompilationManager compilationManager,
+            ISerializer serializer)
+            : base(LangServerConstants.GetDeploymentParametersCommand, serializer)
+        {
+            this.compilationManager = compilationManager;
+        }
+
+        public override Task<BicepDeploymentParametersResponse> Handle(string documentPath, string parametersFilePath, string template, CancellationToken cancellationToken)
```

Comment:
I'm confused.  Why "string template"?  Aren't we working with a bicep file?

Reply:
1. We already have the template generated as soon as deployment is kicked off.
https://github.com/Azure/bicep/blob/d0d6ff82e6ba6657c6e9fb4c739b117f2a3bf404/src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs#L83
2. The template generated above is eventually used in start deployment call here: https://github.com/Azure/bicep/blob/ef256eee93b154803994740f0b34a714d24ac73a/src/Bicep.LangServer/Deploy/DeploymentHelper.cs#L83
3. I am using the template generated in step(1) here for simplicity. If we were to get it from bicep file, it's complicated as we would need to get the value off of the modifier - https://github.com/Azure/bicep/blob/7d2b7818e0bd0432560f46713a8584dc21fa12bc/src/Bicep.Core/Syntax/ParameterDeclarationSyntax.cs#L28 which include "=" sign and may include whitespaces

Reply:
There's a function call to get the default value from a parameter.  But if it works, I guess...

Reply:
See src\Bicep.Core\Analyzers\Linter\Rules\SecureParameterDefaultRule.cs

Reply:
Thanks for the suggestion. I'll investigate and open an issue for refactoring if that's a better approach. I'll leave this one as is. I am hesitant to make the refactoring as part of this change.

Reply:
Yep, makes sense.